### PR TITLE
fix: Qalabel label patch

### PR DESCRIPTION
**Pull request by seshapad**

### CODE REVIEW

File: transformer/kubernetes/apiresource/service.go [hunk 1]
```diff
@@ -360,21 +360,21 @@ func (d *Service) createIngress(ir irtypes.EnhancedIR, targetCluster collecttype
 	if len(hostHTTPIngressPaths) == 0 {
 		return nil
 	}
+	qaLabel := targetCluster.Labels[collecttypes.ClusterQaLabelKey]
+	// QALabel prefix for cluster
+	qaId := common.ConfigTargetKey + common.Delim + qaLabel
 	// Set the default ingressClass value
-	ingressClassName := qaengine.FetchStringAnswer(common.ConfigIngressClassName, "Provide the Ingress class name for ingress", []string{"Leave empty to use the cluster default"}, "")
+	ingressClassName := qaengine.FetchStringAnswer(qaId+common.Delim+common.ConfigIngressClassNameKeySuffix, "Provide the Ingress class name for ingress", []string{"Leave empty to use the cluster default"}, "")
```
Comment (HarikrishnanBalagopal) [inline transformer/kubernetes/apiresource/service.go:367]:
```suggestion
	ingressClassName := qaengine.FetchStringAnswer(common.JoinQASubKeys(qaId, common.ConfigIngressClassNameKeySuffix), "Provide the Ingress class name for ingress", []string{"Leave empty to use the cluster default"}, "")
```

File: transformer/kubernetes/apiresource/service.go [hunk 2]
```diff
@@ -360,21 +360,21 @@ func (d *Service) createIngress(ir irtypes.EnhancedIR, targetCluster collecttype
...
 	secretName := ""
 	defaultSecretName := ""
 	if host == "" {
-		qaLabel := targetCluster.Labels[collecttypes.ClusterQaLabelKey]
 		host = commonqa.IngressHost(d.getHostName(ir.Name), qaLabel)
-		secretName = qaengine.FetchStringAnswer(common.ConfigIngressTLSKey+common.Delim+qaLabel, "Provide the TLS secret for ingress", []string{"Leave empty to use http"}, defaultSecretName)
-	} else {
-		secretName = qaengine.FetchStringAnswer(common.ConfigIngressTLSKey, "Provide the TLS secret for ingress", []string{"Leave empty to use http"}, defaultSecretName)
 	}
+	secretName = qaengine.FetchStringAnswer(qaId+common.Delim+common.ConfigIngressTLSKeySuffix, "Provide the TLS secret for ingress", []string{"Leave empty to use http"}, defaultSecretName)
```
Comment (HarikrishnanBalagopal) [inline transformer/kubernetes/apiresource/service.go:377]:
```suggestion
	secretName = qaengine.FetchStringAnswer(common.JoinQASubKeys(qaId, common.ConfigIngressTLSKeySuffix), "Provide the TLS secret for ingress", []string{"Leave empty to use http"}, defaultSecretName)
```

File: types/qaengine/commonqa/commonqa.go [hunk 3]
```diff
@@ -75,7 +75,7 @@ func ImageRegistryNamespace() string {
 
 // IngressHost returns Ingress host
 func IngressHost(defaulthost string, clusterQaLabel string) string {
-	key := common.ConfigTargetKey + common.Delim + clusterQaLabel + common.Delim + common.ConfigIngressHostKey
+	key := common.ConfigTargetKey + common.Delim + clusterQaLabel + common.Delim + common.ConfigIngressHostKeySuffix
```
Comment (HarikrishnanBalagopal) [inline types/qaengine/commonqa/commonqa.go:78]:
```suggestion
	key := common.JoinQASubKeys(common.ConfigTargetKey, clusterQaLabel, common.ConfigIngressHostKeySuffix)
```

**Review by HarikrishnanBalagopal — CHANGES_REQUESTED** [review]
some minor changes

**Review by ashokponkumar — APPROVED** [review]

**Comment by ashokponkumar** [comment]
@HarikrishnanBalagopal Can you review this PR again, so that it automerges?

**Comment by HarikrishnanBalagopal** [comment]
> @HarikrishnanBalagopal Can you review this PR again, so that it automerges?

oops I requested a review from you accidentally

File: transformer/kubernetes/apiresource/service.go [hunk 4]
```diff
@@ -360,21 +360,21 @@ func (d *Service) createIngress(ir irtypes.EnhancedIR, targetCluster collecttype
 	if len(hostHTTPIngressPaths) == 0 {
 		return nil
 	}
+	qaLabel := targetCluster.Labels[collecttypes.ClusterQaLabelKey]
+	// QALabel prefix for cluster
+	qaId := common.ConfigTargetKey + common.Delim + qaLabel
```
Comment (HarikrishnanBalagopal) [inline transformer/kubernetes/apiresource/service.go:365]:
```suggestion
	qaId := common.JoinQASubKeys(common.ConfigTargetKey, qaLabel)
```

File: transformer/kubernetes/apiresource/service.go [hunk 5]
```diff
@@ -360,21 +360,21 @@ func (d *Service) createIngress(ir irtypes.EnhancedIR, targetCluster collecttype
 	if len(hostHTTPIngressPaths) == 0 {
 		return nil
 	}
+	qaLabel := targetCluster.Labels[collecttypes.ClusterQaLabelKey]
```
Comment (HarikrishnanBalagopal) [inline transformer/kubernetes/apiresource/service.go:363]: will this always succeed? can `qaLabel` be empty string?

**Review by HarikrishnanBalagopal — CHANGES_REQUESTED** [review]
some more minor stuff

File: transformer/kubernetes/apiresource/service.go (hunk 4 above)
Comment (seshapad) [reply transformer/kubernetes/apiresource/service.go:368]: addressed.

File: transformer/kubernetes/apiresource/service.go (hunk 5 above)
Comment (seshapad) [reply transformer/kubernetes/apiresource/service.go]: addressed.

**Review by HarikrishnanBalagopal — APPROVED** [review]
LGTM

**Pull request merged**